### DISCUSSION TITLE
how to change the layer's attributes dynamically

### DISCUSSION THREAD

**Discussion by juzi5230** [visgl/deck.gl#6487] [Q&A]
I want to  modify the color of  GeoJsonLayer dynamically, but failed

        new GeoJsonLayer({
          id: 'HEAT_MAP' ,
          data:  geojson, // geojson is a variable， everytime before render it will be computed again
          opacity: 1,
          stroked: false,
          extruded: true,
          elevationScale: 1,
          lineWidthUnits: 'meters',
          lineWidthMinPixels:1,
          lineWidthMaxPixels:3,
          getElevation: f => Math.sqrt(f.properties.valuePerSqm) * 10,
          getFillColor: d => {
            if(d.properties.num) return [255, 0, 0] // get the attribute of num and get the colorValue
            else return [0, 0, 0, 0]
          },
          getLineWidth: 1,
          getRadius: 10,
          getLineColor: [103, 113, 121],
          pickable: true,
        })

**Comment by linze99** [comment]
```js
function render() {
  const layers = [ new GeoJsonLayer({...})];
  deck.setProps({layers});
}
```

you can use the deck.setProps({layers:[layer]}) method to update the geojson data,notice that you have to pass another object (not the same reference to the old variable) to the method, otherwise it will not trigger the update.
